sdk/core/azure-mgmt-core/azure/mgmt/core/policies/_policy_token_header_async.py

```diff
@@ -0,0 +1,103 @@
+    """Async HTTP pipeline policy for adding policy token headers to Azure Resource Manager requests.
+
+    This policy handles the acquisition and application of Azure Policy tokens for external
+    policy evaluation. It integrates with the Azure Policy service to obtain tokens that
+    enable policy evaluation during resource operations.
+
+    :param client: The async ARM pipeline client used for making policy token requests
+    :type client: ~azure.mgmt.core.AsyncARMPipelineClient
+    :param kwargs: Additional keyword arguments passed to the base policy
+    :type kwargs: Any
+    """
+
+    def __init__(self, client: "AsyncARMPipelineClient", **kwargs: Any) -> None:  # pylint: disable=unused-argument
+        """Initialize the async policy token header policy.
+
+        :param client: The async ARM pipeline client used for making policy token requests
+        :type client: ~azure.mgmt.core.AsyncARMPipelineClient
+        :param kwargs: Additional keyword arguments passed to the base policy
+        :type kwargs: Any
+        """
+        self._client = client
+
+    async def on_request(self, request: PipelineRequest[HTTPRequestType]) -> None:
+        """Process the request to add policy token headers if needed.
+
+        This method is called for each outgoing request. It checks if a policy token
+        is needed and if so, acquires one and adds it to the request headers.
+
+        :param request: The pipeline request to process
+        :type request: ~azure.core.pipeline.PipelineRequest
+        """
+        acquire_policy_token = request.context.options.pop("acquire_policy_token", False)
+        if not acquire_policy_token or request.http_request.method.upper() == "GET":
+            return
+        try:
+            acquire_policy_request = _create_acquire_policy_request(request)
+            acquire_policy_request.url = self._client.format_url(acquire_policy_request.url)
+            acquire_policy_response = await self._client.send_request(acquire_policy_request, stream=False)
+            _update_request_with_policy_token(request, acquire_policy_request, acquire_policy_response)
+        except Exception:
+            request.context.options["acquire_policy_token"] = True
+            raise
+
+    async def send(
+        self, request: PipelineRequest[HTTPRequestType]
+    ) -> PipelineResponse[HTTPRequestType, AsyncHTTPResponseType]:
+        """Authorize request with a bearer token and send it to the next policy.
+
+        This method processes the request by calling on_request to potentially add
+        policy token headers, then forwards the request to the next policy in the pipeline.
+
+        :param request: The pipeline request object
+        :type request: ~azure.core.pipeline.PipelineRequest
+        :return: The pipeline response object
+        :rtype: ~azure.core.pipeline.PipelineResponse
+        """
+        await await_result(self.on_request, request)
```

Comment:
`await_result` isn't needed as we know this will always be an async method.
```suggestion
        await self.on_request(request)
```

Reply:
make sense